src/monitor/node_active_protocol.c

```diff
@@ -1129,6 +1129,14 @@ RemoveNode(AutoFailoverNode *currentNode)
 		if (primaryNode)
 		{
 			(void) ProceedGroupState(primaryNode);
+			LogAndNotifyMessage(
+				message, BUFSIZE,
+				"Setting goal state of %s:%d to apply_settings "
+				"after removing standby node %s:%d from formation %s.",
+				primaryNode->nodeHost, primaryNode->nodePort,
+				currentNode->nodeHost, currentNode->nodePort,
+				formation->formationId);
+			SetNodeGoalState(primaryNode, REPLICATION_STATE_APPLY_SETTINGS, message);
```

Comment:
I don't think we want to do that when `ProceedGroupState` sets the goal state for the primary node to anything other than PRIMARY, such as WAIT_PRIMARY or JOIN_PRIMARY. In those cases, we should also verify that the `synchronous_standby_names` is fetched again from the monitor, but I think it is, at least for WAIT_PRIMARY.